### DISCUSSION TITLE
The API is returning 3 results for every request

### DISCUSSION THREAD

**Discussion by michaeldeathless** [deepgram/community#375] [General help] [answered]
Hello All!

Not sure if this is where I go for support.

I've used some example code from the Deepgram website to take speech streamed from a microphone from my browser, and return text which will be displayed in the browser. For some reason "onmessage" returns 3 different results for each transcription.

So if i say "How was your day?" the result is 3 separate onmessge returns and the result in the console log is :
How was your day? How was your day? How was your day?

Please help! Thanks

The code:

navigator.mediaDevices.getUserMedia({ audio: true }).then((stream) => {
      const mediaRecorder = new MediaRecorder(stream);
      const socket = new WebSocket('wss://api.deepgram.com/v1/listen?smart_format=true&model=nova-2-ea', ['token', 'xxxxxxx']);

      socket.onopen = () => {
        console.log({ event: 'onopen' });
        // Additional logic when socket connection is open
        if (mediaRecorder.state === 'inactive') {
          console.log('addEventListener');
          mediaRecorder.addEventListener('dataavailable', event => {
            if (event.data.size > 0 && socket.readyState == 1) {
              socket.send(event.data);
            }
          });
          mediaRecorder.start(150);
        }
      };

      socket.onmessage = (message) => {
        console.log("onmessage called: ");
        console.log({ event: 'onmessage', message });
        const received = JSON.parse(message.data);
        let transcript = received.channel.alternatives[0].transcript;
       
        if (transcript && received.is_final && transcript.trim() !== '' ) {
          // Set the transcript to the textarea content
          console.log("transcript: ", transcript);
          setContent((prevContent) => prevContent + transcript + ' ');
           
        }
      };

      socket.onclose = () => {
        console.log({ event: 'onclose' });
        // Additional logic when socket connection is closed
      };

      socket.onerror = (error) => {
        console.log({ event: 'onerror', error });
        // Additional logic when socket encounters an error
      };

    
-------

This was returned from the api:

​
message: message { target: WebSocket, isTrusted: true, data: '{"type":"Results","channel_index":[0,1],"duration":1.5,"start":900.81,"is_final":true,"speech_final":true,"channel":{"alternatives":[{"transcript":"How was your day?","confidence":0.9865774,"words":[{"word":"how","start":901.29,"end":901.45,"confidence":0.844791,"punctuated_word":"How"},{"word":"was","start":901.45,"end":901.69,"confidence":0.9783609,"punctuated_word":"was"},{"word":"your","start":901.69,"end":901.85,"confidence":0.99440384,"punctuated_word":"your"},{"word":"day","start":901.85,"end":902.31,"confidence":0.9865774,"punctuated_word":"day?"}]}]},"metadata":{"request_id":"aa1a84b3-a295-4f7c-a267-bb0c822f5d46","model_info":{"name":"2-ea-nova","version":"2023-09-18.22988","arch":"nova-2"},"model_uuid":"79dd3ac5-776e-4fcf-b4fb-0f9968bac445"}}', … }
​​

​
message: message { target: WebSocket, isTrusted: true, data: '{"type":"Results","channel_index":[0,1],"duration":1.3900146,"start":1391.92,"is_final":true,"speech_final":true,"channel":{"alternatives":[{"transcript":"How was your day?","confidence":0.985345,"words":[{"word":"how","start":1392.3201,"end":1392.4801,"confidence":0.90184045,"punctuated_word":"How"},{"word":"was","start":1392.4801,"end":1392.64,"confidence":0.95896655,"punctuated_word":"was"},{"word":"your","start":1392.64,"end":1392.8,"confidence":0.9932288,"punctuated_word":"your"},{"word":"day","start":1392.8,"end":1393.3,"confidence":0.985345,"punctuated_word":"day?"}]}]},"metadata":{"request_id":"f46b445e-7058-4fa8-8eee-414605c075fb","model_info":{"name":"2-ea-nova","version":"2023-09-18.22988","arch":"nova-2"},"model_uuid":"79dd3ac5-776e-4fcf-b4fb-0f9968bac445"}}', … }
​​

​
message: message { target: WebSocket, isTrusted: true, data: '{"type":"Results","channel_index":[0,1],"duration":1.7999268,"start":1391.29,"is_final":true,"speech_final":true,"channel":{"alternatives":[{"transcript":"How was your day?","confidence":0.9872884,"words":[{"word":"how","start":1392.0901,"end":1392.25,"confidence":0.90158796,"punctuated_word":"How"},{"word":"was","start":1392.25,"end":1392.49,"confidence":0.9791942,"punctuated_word":"was"},{"word":"your","start":1392.49,"end":1392.65,"confidence":0.99330574,"punctuated_word":"your"},{"word":"day","start":1392.65,"end":1393.09,"confidence":0.9872884,"punctuated_word":"day?"}]}]},"metadata":{"request_id":"0cd680be-881f-406f-87f6-9f3211a55e05","model_info":{"name":"2-ea-nova","version":"2023-09-18.22988","arch":"nova-2"},"model_uuid":"79dd3ac5-776e-4fcf-b4fb-0f9968bac445"}}', … }
​​

**Comment by DamienDeepgram** [comment]
Try setting the encoding and sample rate of your audio stream.

**Reply by alexbainbridge** [reply]
To what Damien? I am now thinking I might have a related issue. I do set a mime type but not sample rate? What would be a good sample rate for mobile device (iOS / Android) using web sockets?

**Reply by michaeldeathless** [reply, marked as answer]
So it turns out it was because I am running next.js in a dev environment and it loads the project twice. Sorry for the inconvenience.

**Comment by DamienDeepgram** [comment]
> To what Damien? I am now thinking I might have a related issue. I do set a mime type but not sample rate? What would be a good sample rate for mobile device (iOS / Android) using web sockets?

Best to check what encoding and sample rate your audio is. Usually this is documented by the provider somewhere. For static files you can use ffprobe but for a stream its a bit trickier
